sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2491,6 +2491,16 @@ class Dataset[T] private[sql](
     }
   }
 
+  /**
+   * Returns a new Dataset by updating an existing column with metadata.
+   *
+   * @group untypedrel
+   * @since 3.3.0
+   */
+  def withMetadata(columnName: String, metadata: Metadata): DataFrame = {
```

Comment:
shall we use function name `withColumnMetadata` to make it more clear ?

Reply:
That would also work, but I prefer the current shorter version, since the first param requires a column name, it's not likely for the users to be confused about whether the metadata is for a column or something else. Also, since this is syntax sugar, a shorter name would better justify the new API.